module/VuFind/src/VuFind/Content/PageLocator.php

```diff
@@ -84,35 +84,102 @@ public function __construct($themeInfo, $language, $defaultLanguage)
         $this->defaultLanguage  = $defaultLanguage;
     }
 
+    /**
+     * Generate a template from a file search pattern. Examples:
+     * - %pathPrefix%/%pageName%{_%language%} => content/help_en
+     * - %pathPrefix%/%language%/%pageName% => HelpTranslations/en/search
+     *
+     * @param string $pathPrefix Subdirectory where the template should be located
+     * @param string $pageName   Page name
+     * @param string $pattern    Filesystem pattern
+     * @param string $language   Language
+     *
+     * @return string
+     */
+    protected function generateTemplateFromPattern(
+        $pathPrefix, $pageName, $pattern, $language=null
+    ) {
+        $template = $pattern;
+        $template = str_replace('%pathPrefix%', $pathPrefix, $template);
+        $template = str_replace('%pageName%', $pageName, $template);
+        $languagePatternExtended = '"\\{(.*)(%language%)(.*)\\}"';
+        if (isset($language)) {
+            $template = preg_replace(
+                $languagePatternExtended,
+                "\\1$language\\3",
+                $template
+            );
+        } else {
+            $template = preg_replace($languagePatternExtended, '', $template);
+        }
+        $template = str_replace('%language%', $language ?? '', $template);
```

Comment:
If we only allow strings, ` ?? ''` would not be needed.

Reply:
$language currently defaults to `null` in the function signature, and I think passing null as the second parameter of str_replace may trigger a notice... so I think we either need this, or we need to change the function signature to default to something different (like an empty string).

Reply:
Yes, but I proposed making the default an empty string as there's no need to distinguish between null and ''.

Reply:
Ahh, sorry, I was just viewing this comment in isolation. If we change the default, I agree entirely!